### DISCUSSION TITLE
Cannot configure HOST

### DISCUSSION THREAD

**Discussion by zenoverflow** [open-webui/open-webui#5302] [General]
To get access to an ollama instance running on `127.0.0.1`, the container needs to be run with `--network=host`.

That works, but now I have a server listening on `0.0.0.0`, which is not okay on this system for dumb security reasons I cannot disclose.

I looked through the docs and there is no `HOST` env variable, only `PORT`. Setting `UVICORN_HOST` doesn't work either.

Is there no way to configure the host or am I missing something?

PS: Ollama is used by other software on the system, so a workaround by using the docker compose variant with an internal ollama service is a no-go.

**Comment by justinh-rahb** [comment]
Inside a Docker container, `localhost` refers to the container itself, not your computer. To access services on your host from a container, use the Docker bridge network, your computer's IP, or hostname—`localhost` won't work. For external access, your host's services must be listening on the correct interface. By default, Ollama listens on 127.0.0.1. While many suggest using 0.0.0.0, since that's not an option, use the Docker default gateway for the host (172.17.0.1).

**Reply by zenoverflow** [reply]
Good suggestion. Will keep it in mind in the future.

I ended up with a quick bruteforce approach, locking things down via an external firewall.

Best-case scenario, the system should eventually be migrated to docker-compose where this config is a non-issue.
